File: Fall Guys_Final/Fall Guys_Final/Assets/Scripts/Network/Schema/GameState.cs
// Auto-matched to server src/schema/GameState.ts — keep field order in sync.
using Colyseus.Schema;

public class GameState : Schema
{
    [Type(0, "string")]  public string  phase         = default;
    [Type(1, "float32")] public float   timer         = default;
    [Type(2, "uint8")]   public byte    aliveCount    = default;
    [Type(3, "uint8")]   public byte    finishedCount = default;

    [Type(4, "map", typeof(MapSchema<PlayerState>))]
    public MapSchema<PlayerState> players = new MapSchema<PlayerState>();
}

File: server/src/schema/GameState.ts
import { Schema, MapSchema, type } from "@colyseus/schema";
import { PlayerState } from "./PlayerState";

export type GamePhase = "waiting" | "countdown" | "playing" | "roundOver";

export class GameState extends Schema {
    @type("string")  phase: GamePhase = "waiting";
    @type("float32") timer: number = 180;
    @type("uint8")   aliveCount: number = 0;
    @type("uint8")   finishedCount: number = 0;

    @type({ map: PlayerState })
    players = new MapSchema<PlayerState>();
}
